cockroachdb/src/jepsen/cockroach/client.clj

```diff
@@ -224,6 +221,15 @@
         #"restart transaction"
         {:type :fail, :error [:restart-transaction m]}
 
+        ; Explicitly indicates that the operation may or may not succeed.
+        #"result is ambiguous"
+        {:type :info, :error [:result-ambiguous m]}
+
+        ; By default, interpret errors as ambiguous by marking them as :info.
+        ;
+        ; TODO(pavelkalinnikov): do better at detecting "ambiguous" errors
+        ; above. If all the cases are covered above, we can then safely assign
+        ; type :fail to unknown errors by default.
```

Comment:
For my own understanding: how far are we from being able to ensure that every ambiguous error is returned as a `result is ambiguous` error? It sounds to me like it would be better if crdb were able to provide that "contract" instead of refining error detection in this client.

Reply:
You are right that we should ideally catch all ambiguous errors before returning them from CRDB, and return them as "result is ambiguous". I don't know how realistic this is though, maybe a more robust approach is to interpret all errors as ambiguous by default. It seems easier to make a list of errors that mean the request definitely failed, than a list of errors (one catch-all regex ideally) that mean it could fail or succeed.